andela/wg-fortem
# #166152333 Add support for macrocycle, mesocycle and microcycle

#### Title
Adds support for macrocycle(Up to a year), mesocycle(2-6 weeks) and microcycle(1 week).

#### Description
Macro is an annual plan, meso is phase of training with a duration of between 2-6 weeks and micro is instead only a week.

This is needed for sports periodization ( powerlifting, weightlifting, strongman, etc ) or at least to create fitness programs drafted on more than only one week.

#### Type of change






#### How Has This Been Tested?

-  check out the branch 
   `story/166152333-macrocycle-mesocycle-microcycle-support`

-  run the command  
   `nvoke bootstrap-wger --settings-path ./wger/settings.py --no-start-server` and then `python manage.py runserver`.

- run server and open browser
  `python manage.py migrate`

- run server and open browser
  `python manage.py  runserver`

- navigate to workouts page and click on Add workout
![image](https://user-images.githubusercontent.com/11487123/59452600-b8102000-8e16-11e9-967c-df02cb6a69e9.png)

- set a cycle type
![image](https://user-images.githubusercontent.com/11487123/59452697-fc9bbb80-8e16-11e9-8ed8-a74ed10b486e.png)

  select the cycle you want, put a description and save. Depending on which cycle you chose, one of 
  the following should appear and  the popups while setting a timeline

`Microcycle` 
![image](https://user-images.githubusercontent.com/11487123/59453060-c01c8f80-8e17-11e9-940d-7a046f4e851e.png)
![image](https://user-images.githubusercontent.com/11487123/59455309-8e59f780-8e1c-11e9-947e-7666b2b8b474.png)


`Mesocycle`
![image](https://user-images.githubusercontent.com/11487123/59453079-c9a5f780-8e17-11e9-8843-2e6db918c74e.png)
![image](https://user-images.githubusercontent.com/11487123/59453090-ce6aab80-8e17-11e9-8446-df759c57beba.png)

`Macrocycle`
![image](https://user-images.githubusercontent.com/11487123/59453103-d591b980-8e17-11e9-9f9b-460d40f3f3b3.png)
![image](https://user-images.githubusercontent.com/11487123/59453330-5355c500-8e18-11e9-9a32-f1797971f451.png)

#### Checklist:
- [x] Add workout cycle fixtures in weeks and month as well days
- [x] Add a workout_plan field in the core_days_of_week model
- [x] Add cycle options field in the manager_workout model
- [x] Add set cycle type  modal link in the single workout view

#### PT stories
[#166152333](https://www.pivotaltracker.com/story/show/166152333)
